eng/pipelines/performance/perf-build.yml

```diff
@@ -0,0 +1,237 @@
+parameters:
+- name: runPrivateJobs
+  type: boolean
+  default: false
+- name: mauiFramework
+  type: string
+  default: 'net9.0'
+- name: coreclr_arm64_linux
+  displayName: Build Coreclr Arm64 Linux
+  type: boolean
+  default: true
+- name: coreclr_arm64_windows
+  displayName: Build Coreclr Arm64 Windows
+  type: boolean
+  default: true
+- name: coreclr_muslx64_linux
+  displayName: Build Coreclr Musl x64 Linux
+  type: boolean
+  default: true
+- name: coreclr_x64_linux
+  displayName: Build Coreclr x64 Linux
+  type: boolean
+  default: true
+- name: coreclr_x64_windows
+  displayName: Build Coreclr x64 Windows
+  type: boolean
+  default: true
+- name: coreclr_x86_windows
+  displayName: Build Coreclr x86 Windows
+  type: boolean
+  default: true
+- name: coreclr_arm64_android
+  displayName: Build Coreclr Arm64 Android
+  type: boolean
+  default: true
+- name: wasm
+  displayName: Build WebAssembly (wasm)
+  type: boolean
+  default: true
+- name: monoAot_arm64_linux
+  displayName: Build Mono AOT Arm64 Linux
+  type: boolean
+  default: true
+- name: monoAot_x64_linux
+  displayName: Build Mono AOT x64 Linux
+  type: boolean
+  default: true
+- name: mono_x64_linux
+  displayName: Build Mono x64 Linux
+  type: boolean
+  default: true
+- name: mono_arm64_linux
+  displayName: Build Mono Arm64 Linux
+  type: boolean
+  default: true
+- name: mono_arm64_android
+  displayName: Build Mono Arm64 Android
+  type: boolean
+  default: true
+- name: mono_arm64_ios
+  displayName: Build Mono Arm64 iOS
+  type: boolean
+  default: true
+- name: android_arm64
+  displayName: Build Android Arm64 (Not working)
+  type: boolean
+  default: false # currently not working
+- name: ios_arm64
+  displayName: Build iOS Arm64
+  type: boolean
+  default: true
+
+trigger:
+  batch: false # we want to build every single commit
+  branches:
+    include:
+    - main
+    - release/9.0
+    - release/8.0
+  paths:
+    include:
+    - '*'
+    - src/libraries/System.Private.CoreLib/*
+    exclude:
+    - '**.md'
+    - .devcontainer/*
+    - .github/*
+    - docs/*
+    - LICENSE.TXT
+    - PATENTS.TXT
+    - THIRD-PARTY-NOTICES.TXT
+
+resources:
+  repositories:
+    - repository: performance
+      type: git
+      name: internal/dotnet-performance
+      ref: ekharion/new-perf-builds-pipeline # remove before merging
+
+variables:
+  - template: /eng/pipelines/common/variables.yml
+
+extends:
+  template: /eng/pipelines/common/templates/pipeline-with-resources.yml
+  parameters:
+    stages:
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), or(eq(variables['Build.Reason'], 'Schedule'), parameters.runPrivateJobs)) }}:
```

Comment:
@LoopedBard3 @caaavik-msft Is checking against `public` still correct? I think the name of the project where our public CI runs go is now called `dnceng-public`